File: 2021/09/21/main.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        r, i, cnt = 0, 0, 0
        while i < len(nums):
            while i < len(nums) and nums[i] == 1:
                i += 1
                cnt += 1
            r = max(r, cnt)
            i += 1
            cnt = 0
        return r

File: 2021/09/21/main.go
package main

func findMaxConsecutiveOnes(nums []int) int {
	r, i, cnt := 0, 0, 0
	for i < len(nums) {

		for i < len(nums) && nums[i] == 1 {
			i += 1
			cnt += 1
		}
		r = max(r, cnt)
		i += 1
		cnt = 0
	}
	return r
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
